isyfact-standards-doc/src/docs/antora/modules/referenzarchitektur/pages/software-technisch/services.adoc

```diff
@@ -0,0 +1,247 @@
+====
+Weitere Details zu JMS-Nachrichten finden sich in der JMS-Spezifikation im Kapitel https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-model[3. Jakarta Messaging message model].
+Besonders relevant für die Referenzarchitektur sind die Abschnitte https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-messages[3.3. Jakarta Messaging messages] sowie https://jakarta.ee/specifications/messaging/3.0/jakarta-messaging-spec-3.0.html#jakarta-messaging-message-body[3.11. Jakarta Messaging message body].
+====
+
+== Kommunikation mit externen Systemen
+
+Die Kommunikation mit externen Systemen basiert auf Web-Services.
+Wird ein Service von einem externen System angeboten, wird er als *externer Service* bezeichnet.
+Hierbei muss man zwischen zwei Fällen unterscheiden:
+
+*Aufruf von Services der Systemlandschaft*: Durch die Systemlandschaft wird externen Systemen die Schnittstelle eines Backends in Form eines Web-Services zur Verfügung gestellt.
+Hierbei definiert das Backend selbst keinen Web-Service.
+Vielmehr definiert das Backend, wie bei der internen Kommunikation auch, eine Schnittstelle.
+Diese Schnittstelle wird dann durch ein eigenständiges IT-System als Web-Service exportiert.
+Dieses IT-System wird als *Service-Provider* bezeichnet.
+Für jede Schnittstelle, die als Web-Services exportiert werden soll, muss ein eigener Service-Provider definiert werden.
+
+*Nutzung von externen Services*: Ähnlich wie im vorigen Fall ruft das interne IT-System den externen Service nicht direkt auf.
+Es ruft ein eigenständiges IT-System auf, welches den externen Service als Schnittstelle in die Systemlandschaft importiert.
+Dieses IT-System wird als *Service-Consumer* bezeichnet.
+Das interne IT-System ruft dann lediglich die Schnittstelle des Service-Consumers auf.
+Für das interne IT-System ist dieser Aufruf nicht von einem Aufruf zu einem anderen internen IT-System zu unterscheiden.
+Für jeden Web-Service, der in die Systemlandschaft importiert werden soll, muss ein eigener Service-Consumer definiert werden.
+
+Die Gesamtheit aller Service-Provider und Service-Consumer wird als xref:glossary:glossary:master.adoc#glossar-service-gateway[Service-Gateway] bezeichnet.
+Die Service-Gateways stellen somit die zentrale Schnittstelle einer IsyFact-Systemlandschaft zur Außenwelt dar.
+
+Ein Service-Consumer macht diesen "externen Service" als "inneren Service" der Systemlandschaft verfügbar.
+Wird ein Service von einem Backend angeboten, so ist das ebenfalls ein "innerer Service".
+Wenn ein Service-Provider diesen "inneren Service" einer Anwendung außerhalb der Plattform zugänglich macht, ist dies ein "äußerer Service" der Systemlandschaft.
+Die Unterscheidung zwischen "innere" und "äußere" ist analog für die Begriffe "Request" und "Response" zu verwenden.
+
+image::software-technisch/services/kommunikation-externe-systeme.dn.svg[]
+
+[[versionierung]]
+== Versionierung
```

Comment:
Vielleicht sollte man irgendwo noch klären, wann überhaupt eine neue Service-Version benötigt wird.
Bei aufsteigend kompatiblen Änderungen der Schnittstelle (Schnittstelle wurde erweitert um neue Operationen, oder Schnittstellen-Implementierung wurde verallgemeinert, sodass sie sowohl die alte wie auch die neue Signatur einer Operation unterstützt) 
=> bei Änderungen dieser Art benötigt man keine neue Version.

Reply:
Dieses Thema wird in einem Folgeticket behandelt.